presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -414,6 +414,24 @@ public PlanWithProperties visitTopN(TopNNode node, PreferredProperties preferred
         @Override
         public PlanWithProperties visitSort(SortNode node, PreferredProperties preferredProperties)
         {
+            if (isDistributedSortEnabled(session)) {
+                PlanWithProperties child = planChild(node, PreferredProperties.any());
+                if (!child.getProperties().isSingleNode()) {
```

Comment:
there is a slight difference here where we skip distributed sort if the child is already on a single node.  this is similar to the logic that was happening when this check came after the isSingleNode check below but not exactly the same (the child is different because it was planned with the distributed sort preferred properties and also doesn't depend on whether the data was already sorted -- though I can certainly add that if that's preferred)

Reply:
cc @sopel39